python/src/spark_rapids_ml/umap.py

```diff
@@ -1396,14 +1593,46 @@ def load(self, path: str) -> "_CumlEstimator":
         model_attr_str = self.sc.textFile(metadata_file_path).collect()[0]
         model_attr_dict = json.loads(model_attr_str)
 
+        spark = _get_spark_session()
+
         for key, value in model_attr_dict.items():
-            if isinstance(value, list) and value[0].endswith(".npy"):
-                arrays = []
-                spark = _get_spark_session()
-                for array_path in value:
-                    array = np.load(array_path)
-                    arrays.append(spark.sparkContext.broadcast(array))
-                model_attr_dict[key] = arrays
+            if isinstance(value, str) and value.endswith(".npz"):
+                npz_data = np.load(value)
+                if value.endswith("csr_.npz"):
+                    # Handle sparse raw data - npz file contains separate arrays for CSR attributes
+                    def _get_sorted_data_keys(
+                        npz_data: np.lib.npyio.NpzFile, csr_key: str
+                    ) -> List[str]:
+                        # Filter keys that start with the given csr_key and sort by the chunk index
+                        sorted_keys = sorted(
+                            [key for key in npz_data.keys() if key.startswith(csr_key)],
+                            key=lambda x: int(x.split("_")[1]),
+                        )
+                        return sorted_keys
+
+                    indices_keys = _get_sorted_data_keys(npz_data, "indices")
+                    indptr_keys = _get_sorted_data_keys(npz_data, "indptr")
+                    data_keys = _get_sorted_data_keys(npz_data, "data")
+                    raw_data_dict = {
+                        "indices": [
+                            spark.sparkContext.broadcast(npz_data[key])
```

Comment:
Related to above comment, seems odd to create broadcast upon loading the model from store.